lib/charms/mongodb/v0/mongodb_backups.py

```diff
@@ -357,3 +381,117 @@ def _get_pbm_status(self) -> StatusBase:
+                    backup_status = "failed: " + backup.get("error", "N/A")
+            if backup["status"] not in ["error", "done"]:
+                backup_status = "in progress"
+            backup_list.append((backup["name"], backup["type"], backup_status))
+
+        # process in progress backups
+        running_backup = pbm_status["running"]
+        if running_backup.get("type", None) == "backup":
+            # backups are sorted in reverse order
+            last_reported_backup = backup_list[0]
+            # pbm will occasionally report backups that are currently running as failed, so it is
+            # necessary to correct the backup list in this case.
+            if last_reported_backup[0] == running_backup["name"]:
+                backup_list[0] = (last_reported_backup[0], last_reported_backup[1], "in progress")
+            else:
+                backup_list.append((running_backup["name"], "logical", "in progress"))
+
+        # sort by time and return formatted output
+        return self._format_backup_list(sorted(backup_list, key=lambda pair: pair[0]))
+
+    def _format_backup_list(self, backup_list) -> str:
+        """Formats provided list of backups as a table."""
+        backups = ["{:<21s} | {:<12s} | {:s}".format("backup-id", "backup-type", "backup-status")]
+
+        backups.append("-" * len(backups[0]))
+        for backup_id, backup_type, backup_status in backup_list:
+            backups.append(
+                "{:<21s} | {:<12s} | {:s}".format(backup_id, backup_type, backup_status)
+            )
+
+        return "\n".join(backups)
+
+    def _backup_from_different_cluster(self, backup_status: str) -> bool:
+        """Returns if a given backup was made on a different cluster."""
+        return re.search(REMAPPING_PATTERN, backup_status) is not None
+
+    def _try_to_restore(self, backup_id: str, event) -> None:
+        try:
+            remapping_args = self._remap_replicaset(backup_id)
+            self.charm.run_pbm_restore_command(backup_id, remapping_args)
+            event.set_results({"restore-status": "restore started"})
+            self.charm.unit.status = MaintenanceStatus("restore started/running")
+        except (subprocess.CalledProcessError, ExecError) as e:
+            if type(e) == subprocess.CalledProcessError:
+                error_message = e.output.decode("utf-8")
+            else:
+                error_message = str(e.stderr)
+            fail_message = f"Failed to restore MongoDB with error: {str(e)}"
+
+            if "Resync" in error_message:
+                logger.error("PBM is resyncing %s.", fail_message)
+                raise PBMBusyError
+
+            if f"backup '{backup_id}' not found" in error_message:
+                fail_message = f"Backup id: {backup_id} does not exist in list of backups, please check list-backups for the available backup_ids."
+
+            event.fail(fail_message)
+
+    def _remap_replicaset(self, backup_id: str) -> str:
```

Comment:
This function is taken from VM charm with minimal changes (only in how to run command)